provider/lxd/environ_network_test.go

```diff
@@ -241,6 +241,13 @@ func (s *environNetSuite) TestSubnetDiscoveryFallbackForOlderLXDs(c *gc.C) {
 	env := s.NewEnviron(c, srv, nil).(*environ)
 
 	ctx := context.NewCloudCallContext()
+
+	// Spaces should be supported
+	supportsSpaces, err := env.SupportsSpaces(ctx)
```

Comment:
This bit was why I didn't catch this yesterday. The original implementation just returned "true, nil" so there was no test for it. I added the checks for Subnets() but forgot to add one for SupportsSpaces (which is the one that triggers the NPE when you run `juju spaces`)